### PR TITLE
[M] Upgrade commons-lang3 and liquibase

### PR DESCRIPTION
**Pull request by nikosmoum**
- Now that the commons-lang3 bug (in RandomStringUtils.random()) has been fixed (https://issues.apache.org/jira/browse/LANG-1748) and PNC gradle builds honor the 'strictly' directive, we can upgrade these 2 libraries.

### CODE REVIEW

**Comment by Ceiu** [comment]
@joshmalbrecht Do you happen to remember what you did to hit this? I don't recall if we had specific testing around it or not.

**Comment by nikosmoum** [comment]
The cause for this is described here: https://bugzilla.redhat.com/show_bug.cgi?id=2300323#c1

**Comment by Ceiu** [comment]
nack

Looks like we're still busted here if we upgrade. Testing this branch on Alma (or any other distro that doesn't come stock with some entropy gathering utility for container/VM envs), as soon as the `/dev/random` source is exhausted, CP cannot startup and will hang as we saw in the past.

This can be tested in our current environment as follows:
- create a new el8 vagrant instance (main and this branch currently use almalinux for el8)
- Open two terminals within this el8 instance
- In one terminal, run the following command until it blocks waiting for input: `time cat /dev/random | head -c 4096 > /dev/null`. Note that the "time" is optional, but makes the output more clear
- In the other terminal, deploy Candlepin from this branch

Candlepin will appear to hang as was observed before the version lock was put in place. It becomes unstuck by installing any tooling that can be used to supply entropy, like `rng-tools` and its `rngd` service. Likewise, the above `cat` command is also unblocked by its presence.

In a nutshell, we're still blocked if `/dev/random` has no available data to read.

**Comment by Ceiu** [comment]
And just to double check I had the right dependencies in place:

```
(22:11:44) vagrant@candlepin-el8:~/devel/candlepin (pr/4944) $ ./gradlew dependencies | grep liquibase
+--- org.liquibase:liquibase-core:4.30.0
+--- org.liquibase:liquibase-core:4.30.0 (n)
+--- org.liquibase:liquibase-core:4.30.0
+--- org.liquibase:liquibase-core:4.30.0
+--- com.mattbertolini:liquibase-slf4j:5.1.0
+--- com.mattbertolini:liquibase-slf4j:5.1.0 (n)
+--- org.liquibase:liquibase-core:4.30.0
+--- com.mattbertolini:liquibase-slf4j:5.1.0
(22:11:55) vagrant@candlepin-el8:~/devel/candlepin (pr/4944) $ ./gradlew dependencies | grep commons-lang
|    |    +--- org.apache.commons:commons-lang3:3.8.1
|    |    |    \--- org.apache.commons:commons-lang3:3.7 -> 3.8.1
+--- org.apache.commons:commons-lang3:3.17.0
|    |    +--- org.apache.commons:commons-lang3:3.13.0 -> 3.17.0
|    |    |    \--- org.apache.commons:commons-lang3:3.14.0 -> 3.17.0
|    +--- org.apache.commons:commons-lang3:3.17.0
|    +--- org.apache.commons:commons-lang3:3.17.0
|    |    +--- org.apache.commons:commons-lang3:3.14.0 -> 3.17.0
+--- org.apache.commons:commons-lang3:3.17.0 (n)
+--- org.apache.commons:commons-lang3:3.17.0
|    |    +--- org.apache.commons:commons-lang3:3.13.0 -> 3.17.0
|    |    |    \--- org.apache.commons:commons-lang3:3.14.0 -> 3.17.0
|    +--- org.apache.commons:commons-lang3:3.17.0
|    +--- org.apache.commons:commons-lang3:3.17.0
|    |    +--- org.apache.commons:commons-lang3:3.14.0 -> 3.17.0
+--- org.apache.commons:commons-lang3:3.17.0
|    |    +--- org.apache.commons:commons-lang3:3.13.0 -> 3.17.0
|    |    |    \--- org.apache.commons:commons-lang3:3.14.0 -> 3.17.0
|    +--- org.apache.commons:commons-lang3:3.17.0
|    +--- org.apache.commons:commons-lang3:3.17.0
|    |    +--- org.apache.commons:commons-lang3:3.14.0 -> 3.17.0
+--- org.apache.commons:commons-lang3:3.17.0
|    |    +--- org.apache.commons:commons-lang3:3.13.0 -> 3.17.0
|    |    |    \--- org.apache.commons:commons-lang3:3.14.0 -> 3.17.0
|    +--- org.apache.commons:commons-lang3:3.17.0
|    +--- org.apache.commons:commons-lang3:3.17.0
|    |    +--- org.apache.commons:commons-lang3:3.14.0 -> 3.17.0
(22:12:06) vagrant@candlepin-el8:~/devel/candlepin (pr/4944) $ 
```

**Comment by nikosmoum** [comment]
weird. this seems to have been patched in both commons-lang3 and liquibase, and seems to work for everyone else :/
